### PR TITLE
Alps dev

### PR DESCRIPTION
**Pull request by alpana-hep**
This pull request contains the process & run cards for the MC process 'ZNuNuGamma_MonoPhoton-PtG40to130_NLO_fxfx' (Ultra legacy samples) at the NLO. These cards are created using the cards for Znunugamma_Monophoton_Ptg130_NLO_fxfx process but with the updated only the photon pT (40-130) -
[Znunugamma_Monophoton_Ptg130_NLO_fxfx](https://github.com/alpana-hep/genproductions/tree/Alps_dev/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg130_NLO_fxfx)

Instructions I used to create these cards and corresponding gridpacks (just in case it is needed):
Prepare the process card, run card (or other cards which you want to use)
I have copied the process/run card for Znunugamma_Monophoton_Ptg130_NLO  and changed the name of the file & process - also changed the photon pT range.
Once done, run the following instructions
```
git clone https://github.com/cms-sw/genproductions.git (copy the master branch)
cd genproductions/bin/MadGraph5_aMCatNLO/
cp your run/process cards to a directory in the path cards/production/13TeV and name of the directory should be same as the process name in process card
./gridpack_generation.sh <name of process card without _proc_card.dat> <folder containing cards relative to current location>
```

Please check and let me know if anything I need to update.

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_proc_card.dat [hunk 1]
```diff
@@ -0,0 +1,19 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_proc_card.dat:6]: these lines are not needed

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import model loop_sm-no_b_mass
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+
+define l- = e- mu-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+define p = p b b~
+define j = j b b~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_proc_card.dat:15]: these lines are not needed

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,153 @@
+#***********************************************************************
+# Tag name for the run (one word)                                      *
+#***********************************************************************
+  tag_1 =  run_tag  ! name of the run
+#***********************************************************************
+# Number of LHE events (and their normalization) and the required      *
+# (relative) accuracy on the Xsec.                                     *
+# These values are ignored for fixed order runs                        *
+#***********************************************************************
+        100 = nevents
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:10]: change 100 to a larger number, i.e., 2000

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  0.001 =  req_acc  ! Required accuracy (-1=auto determined from nevents
+  25    =  nevt_job ! Max number of events per job in event generation.
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:12]: change 25  to -1 here

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+# These numbers are ignored except if req_acc_FO is equal to -1
+  5000  =  npoints_FO_grid   ! number of points to setup grids
+
+  4     =  niters_FO_grid    ! number of iter. to setup grids
+  10000 =  npoints_FO        ! number of points to compute Xsec
+  6     =  niters_FO         ! number of iter. to compute Xsec
+#***********************************************************************
+# Random number seed                                                   *
+#***********************************************************************
+       4533 =  iseed
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:33]: change to default value 0

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+#***********************************************************************
+  1     =  lpp1     ! beam 1 type (0 = no PDF)
+  1     =  lpp2     ! beam 2 type (0 = no PDF)
+  6500.0        =  ebeam1   ! beam 1 energy in GeV
+  6500.0        =  ebeam2   ! beam 2 energy in GeV
+#***********************************************************************
+# PDF choice: this automatically fixes also alpha_s(MZ) and its evol.  *
+#***********************************************************************
+  lhapdf        =  pdlabel    ! PDF set
+  306000 = lhaid
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:45]: please change line 45 to following two lines:
 $DEFAULT_PDF_SETS = lhaid ! if pdlabel=lhapdf, this is the lhapdf number
 $DEFAULT_PDF_MEMBERS = reweight_PDF

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+# For PDF uncertainty: use LHAPDF with supported set                   *
+#***********************************************************************
+  True  =  reweight_scale    ! reweight to get scale dependence
+  0.5   =  rw_Rscale_down    ! lower bound for ren scale variations
+  2.0   =  rw_Rscale_up      ! upper bound for ren scale variations
+  0.5   =  rw_Fscale_down    ! lower bound for fact scale variations
+  2.0   =  rw_Fscale_up      ! upper bound for fact scale variations
+  False  =  reweight_PDF      ! reweight to get PDF uncertainty
+#  292201        =  PDF_set_min       ! First of the error PDF sets
+#  292302        =  PDF_set_max       ! Last of the error PDF sets
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:87]: these can be removed

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+  0.0   =  drll     ! Min distance between opposite sign lepton pairs
+  0.0   =  drll_sf  ! Min distance between opp. sign same-flavor lepton pa
+  0.0   =  mll      ! Min inv. mass of all opposite sign lepton pairs
+  30.0  =  mll_sf   ! Min inv. mass of all opp. sign same-flavor lepton p
+#***********************************************************************
+# Photon-isolation cuts, according to hep-ph/9801442                   *
+# When ptgmin=0, all the other parameters are ignored                  *
+#***********************************************************************
+  40.0   =  pta     ! Min photon transverse momentum
+  130.0  = ptamax     ! Max photon transverse momentum
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:138]: you should use "40.0 = ptgmin" in the run card, and implement the ptgmax cut in the cuts.f, e.g., https://github.com/alpana-hep/genproductions/blob/Alps_dev/bin/GenValidation/updated_validation_cards/dy_fxfx_mll50/dy012jfxfx_ptbinned/Zpt-0To80_cuts_template.f

Comment (alpana-hep) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_run_card.dat:138]: Hi ,
This I tried but I could not add it successfully given I am not at all familiar with fortran. So I need some inputs on this cuts.f file.

**Comment by alpana-hep** [comment]
Hi 
I have made the changes suggested by you. I added a .f file as well and on these lines I have replaced ptgmin by ptgmax. 
https://github.com/alpana-hep/genproductions/blob/Alps_dev/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx/Znunugamma_Monophoton_Ptg40to130_NLO_fxfx_cuts.f#L316
Let me know if it is fine now.

**Comment by menglu21** [comment]
Hi @alpana-hep please keep the ptgmin in the cuts.f as it was. you just need to add additional part for the ptgmax selection, i.e., add following code
if (ptg.gt.130d0)then
   passcuts_user=.false.
   return
endif

after 
c Loop over all photons
            j=j+1
            
            ptg=pt(pgamma(0,j))
            if(ptg.lt.ptgmin)then
               passcuts_user=.false.
               return
            endif

**Comment by alpana-hep** [comment]
Hi @menglu21 
Thanks for the help :)
I have made the changes as per your suggestions. Please let me know if somehow I missed something.

**Comment by menglu21** [comment]
Hi @alpana-hep actually you don't need the zpt cut part, it's just a reference for you. could you please use the default cuts.f and only modify the pta part as I mentioned above

**Comment by sihyunjeon** [comment]
@menglu21 This is actually related to the issue I just talked to you about, ptG relaxation

**Comment by alpana-hep** [comment]
> cuts.f
Hi, sure I will remove the ZpT cut. I need a clarification on the rest part of your remark -
By 'default cut.f' you meant that I should implement all the cuts defined in run_card.dat for this process in cut.f file?
Or is there somewhere this default_cut.f file available?

I am sorry I am very new to all this and I could not find such 'default_cut.f' file.

**Comment by alpana-hep** [comment]
Hi @menglu21 
Thanks, I found the default cut.f file and have added to this repo. Please check. For now I have kept the file name as like other card.dat file. Let me know if it is needed to be changed

**Pull request merged**